### PR TITLE
feature-erms-2171

### PR DESCRIPTION
**Pull request by anjaalbinnrw**

### CODE REVIEW

File: app/grails-app/controllers/com/k_int/kbplus/AddressController.groovy [hunk 1]
```diff
@@ -60,61 +54,72 @@ class AddressController extends AbstractDebugController {
...
+                break
+            case RDStore.ADRESS_TYPE_LIBRARY:
+                modalId = "addressFormModalLibraryAddress"
+                break
+        }
+        Map model = [
             addressInstance: addressInstance,
-            editable: addressbookService.isAddressEditable(addressInstance, springSecurityService.getCurrentUser())
-        ] // TODO
+            orgId: addressInstance?.org?.id,
```
Comment (klodav) [inline app/grails-app/controllers/com/k_int/kbplus/AddressController.groovy:80]: addressInstance is not null -> ?

File: app/grails-app/views/organisation/_breadcrumb.gsp [hunk 2]
```diff
@@ -15,8 +15,9 @@
                 <semui:crumb message="menu.my.consortia" controller="myInstitution" action="manageMembers" params="[comboType:'Consortium']"/>
                 <semui:crumb text="${orgInstance.getDesignation()}" class="active"/>
             </g:if>
+
             <g:else>
-                <semui:crumb text="${orgInstance.getDesignation()}" class="active"/>
+                <semui:crumb text="${orgInstance.getDesignation()}" class="active"/>or
```
Comment (klodav) [inline app/grails-app/views/organisation/_breadcrumb.gsp:20]: or ?

File: app/grails-app/views/templates/cpa/_address.gsp [hunk 3]
```diff
@@ -4,44 +4,45 @@
             <i class="ui icon building map marker alternate la-list-icon"></i>
         </span>
         <div class="content la-space-right">
-            <g:link controller="address" action="show" id="${address?.id}">
-                <strong>${address.type?.getI10n('value')}:</strong>
+            <strong>${address.type?.getI10n('value')}:</strong>
+            <div class="item" data-href="#addressFormModal${address.id}" data-semui="modal" >
                 <g:if test="${address?.name}">
                     <br />
                     ${address?.name}
```
Comment (klodav) [inline app/grails-app/views/templates/cpa/_address.gsp:11]: address is not null here -> ? .. and several times below

File: app/grails-app/views/address/list.gsp [hunk 4]
```diff
@@ -68,8 +68,8 @@
 								<g:link action="show" id="${addressInstance.id}" class="ui tiny button">${message('code':'default.button.show.label')}</g:link>
 								<g:link action="edit" id="${addressInstance.id}" class="ui tiny button">${message('code':'default.button.edit.label')}</g:link>
 							</td>
-						</tr>
 					</g:each>
+						</tr>
```
Comment (klodav) [inline app/grails-app/views/address/list.gsp:72]: incorrect nesting

File: app/grails-app/views/address/_formModal.gsp [hunk 5]
```diff
@@ -1,9 +1,47 @@
-<%@ page import="com.k_int.kbplus.Address;com.k_int.kbplus.RefdataValue;com.k_int.kbplus.RefdataCategory;de.laser.helper.RDConstants" %>
-
-<semui:modal id="${modalId ?: 'addressFormModal'}"
-             text="${message(code: 'default.add.label', args: [message(code: 'person.address.label')])}">
-
-    <g:form id="create_address_${modalId}" class="ui form" url="[controller: 'address', action: 'create']" method="POST">
+<%@ page import="de.laser.helper.RDStore; com.k_int.kbplus.Address;com.k_int.kbplus.RefdataValue;com.k_int.kbplus.RefdataCategory;de.laser.helper.RDConstants" %>
+<g:if test="${addressId}">
+    <g:set var="addressInstance" value="${com.k_int.kbplus.Address.get(addressId)}"/>
```
Comment (klodav) [inline app/grails-app/views/address/_formModal.gsp:3]: unnecessary qualified reference

File: app/grails-app/views/address/_formModal.gsp [hunk 6]
```diff
@@ -109,31 +156,37 @@
...
-
-            <g:if test="${modalId == 'addressFormModalLegalPatronAddress'}">
-                <input id="type_${modalId}" name="type.id" type="hidden" value="${com.k_int.kbplus.RefdataValue.getByValueAndCategory('Legal patron address', RDConstants.ADDRESS_TYPE)?.id}"/>
+                <input id="type_${modalId}" name="type.id" type="hidden" value="${RDStore.ADRESS_TYPE_POSTAL?.id}"/>
             </g:if>
-
-        </g:if>
-        <g:else>
+            <g:elseif test="${modalId == 'addressFormModalBillingAddress'}">
+                <input id="type_${modalId}" name="type.id" type="hidden" value="${RDStore.ADRESS_TYPE_BILLING?.id}"/>
```
Comment (klodav) [inline app/grails-app/views/address/_formModal.gsp:172]: never use ? to check constants .. several times below

File: app/grails-app/controllers/com/k_int/kbplus/AddressController.groovy (hunk 1 above)
Comment (anjaalbinnrw) [reply app/grails-app/controllers/com/k_int/kbplus/AddressController.groovy:80]: ok

File: app/grails-app/views/organisation/_breadcrumb.gsp (hunk 2 above)
Comment (anjaalbinnrw) [reply app/grails-app/views/organisation/_breadcrumb.gsp:20]: Da sass der Focus wohl falsch, also nicht im Suchfenster.

**Pull request merged**